python/rmm/_lib/memory_resource.pyx

```diff
@@ -135,6 +141,20 @@ cdef class CudaMemoryResource(DeviceMemoryResource):
         pass
 
 
+cdef class CudaAsyncMemoryResource(DeviceMemoryResource):
+    def __cinit__(self, device=None):
+        self.c_obj.reset(
+            new cuda_async_memory_resource()
+        )
+
+    def __init__(self, device=None):
+        """
+        Memory resource that uses cudaMallocAsync/Free for
+        allocation/deallocation
+        """
+        pass
```

Comment:
Is this needed? I think we can just use `__cinit__` if that's all we want

Reply:
It's the only way to make the docstring available I think (the alternative being a class docstring) :(

Reply:
Maybe using the class docstring is fine? There's not too much going on in this class

Reply:
Seems like other libraries like Pandas / PyArrow use class docstrings over `__init__` docstrings anyway.

Reply:
Fixed